### PR TITLE
Updating to latest references + minor fixes

### PR DESCRIPTION
**Pull request by nemakam**
## Description

1. Updated references to latest amqp and other packages. #336 
1. Updated resource string for exception. #375 
1. Added retry logic for cbs token renewal to make it more resilient to network issues. #378 
1. Updated bug where `CbsTokenProvider` was not initialized properly.  #359 (Thanks @Aleksanderis)
1. Support for custom token providers which was added for few clients has been extended to `MessageSender`, `MessageReceiver` and `SessionClient`
1. Minor cleanup.

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs [hunk 1]
```diff
@@ -58,46 +60,51 @@ public void SetActiveRequestResponseLink(ActiveRequestResponseLink requestRespon
...
-                    activeClientLinkObject.Audience,
-                    activeClientLinkObject.Audience,
-                    activeClientLinkObject.RequiredClaims,
-                    ActiveClientLinkManager.SendTokenTimeout).ConfigureAwait(false);
-
-                this.SetRenewCBSTokenTimer(activeClientLinkObject);
+                var renewTokenTask = this.retryPolicy.RunOperation(
+                    async () =>
+                    {
+                        activeClientLinkObject.AuthorizationValidUntilUtc = await cbsLink.SendTokenAsync(
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs]: `ConfigureAwait(false)`

File: src/Microsoft.Azure.ServiceBus/Resources.resx [hunk 2]
```diff
@@ -118,7 +118,7 @@
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
   <data name="AmqpFieldSessionId" xml:space="preserve">
-    <value>No session-id was specified for a session receiver.</value>
+    <value>Failed to retreive session ID. Please retry.</value>
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Resources.resx]: From broker?

File: src/Microsoft.Azure.ServiceBus/Resources.resx [hunk 3]
```diff
@@ -208,7 +208,7 @@
     <value>The {0} plugin has already been registered.</value>
   </data>
   <data name="SessionFilterMissing" xml:space="preserve">
-    <value>Session filter is missing on the link. </value>
+    <value>Failed to retreive session filter. Please retry.</value>
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Resources.resx]: From broker?

File: src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs [hunk 4]
```diff
@@ -76,28 +76,52 @@ public class MessageSender : ClientEntity, IMessageSender
...
+        /// </summary>
+        /// <param name="endpoint">Fully qualified domain name for Service Bus. Most likely, {yournamespace}.servicebus.windows.net</param>
+        /// <param name="entityPath">Queue path.</param>
+        /// <param name="tokenProvider">Token provider which will generate security tokens for authorization.</param>
+        /// <param name="transportType">Transport type.</param>
+        /// <param name="retryPolicy">Retry policy for queue operations. Defaults to <see cref="RetryPolicy.Default"/></param>
+        /// <remarks>Creates a new connection to the entity, which is opened during the first operation.</remarks>
+        public MessageSender(
+            string endpoint,
+            string entityPath,
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs]: null checks for these as well

File: src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs [hunk 5]
```diff
@@ -103,15 +103,44 @@ public class MessageReceiver : ClientEntity, IMessageReceiver
...
+        /// <param name="tokenProvider">Token provider which will generate security tokens for authorization.</param>
+        /// <param name="transportType">Transport type.</param>
+        /// <param name="receiveMode">Mode of receive of messages. Defaults to <see cref="ReceiveMode"/>.PeekLock.</param>
+        /// <param name="retryPolicy">Retry policy for queue operations. Defaults to <see cref="RetryPolicy.Default"/></param>
+        /// <param name="prefetchCount">The <see cref="PrefetchCount"/> that specifies the upper limit of messages this receiver
+        /// will actively receive regardless of whether a receive operation is pending. Defaults to 0.</param>
+        /// <remarks>Creates a new connection to the entity, which is opened during the first operation.</remarks>
+        public MessageReceiver(
+            string endpoint,
+            string entityPath,
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: null checks on these too..

File: src/Microsoft.Azure.ServiceBus/SessionClient.cs [hunk 6]
```diff
@@ -102,8 +103,45 @@ public sealed class SessionClient : ClientEntity, ISessionClient
...
+        /// <param name="tokenProvider">Token provider which will generate security tokens for authorization.</param>
+        /// <param name="transportType">Transport type.</param>
+        /// <param name="receiveMode">Mode of receive of messages. Defaults to <see cref="ReceiveMode"/>.PeekLock.</param>
+        /// <param name="retryPolicy">Retry policy for queue operations. Defaults to <see cref="RetryPolicy.Default"/></param>
+        /// <param name="prefetchCount">The <see cref="PrefetchCount"/> that specifies the upper limit of messages this receiver
+        /// will actively receive regardless of whether a receive operation is pending. Defaults to 0.</param>
+        /// <remarks>Creates a new connection to the entity, which is opened during the first operation.</remarks>
+        public SessionClient(
+            string endpoint,
+            string entityPath,
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/SessionClient.cs]: null checks

**Comment by vinaysurya** [comment]
>     sealed class ActiveClientLinkManager

is there a test we can add for the automatic token renewal? May be 'internal' expose the timelimit for a SAS token, set it to a low value, let it expire and see that renew works

---
Refers to: src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs:11 in 1276eb9. [](commit_id = 1276eb971ffa652f732cb2841580f3ca47aa620c, deletion_comment = False)

File: src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs (hunk 1 above)
Comment (nemakam) [reply src/Microsoft.Azure.ServiceBus/Amqp/ActiveClientLinkManager.cs]: Yup.. Already doing it :)

**Pull request merged**